internal/plugins/helm/v1/api.go

```diff
@@ -15,189 +15,131 @@
-	"github.com/operator-framework/operator-sdk/internal/plugins/manifests"
-	manifestsv2 "github.com/operator-framework/operator-sdk/internal/plugins/manifests/v2"
 )
 
+const (
+	crdVersionFlag       = "crd-version"
+	helmChartFlag        = "helm-chart"
+	helmChartRepoFlag    = "helm-chart-repo"
+	helmChartVersionFlag = "helm-chart-version"
+)
+
+var _ plugin.CreateAPISubcommand = &createAPISubcommand{}
+
 type createAPISubcommand struct {
-	config        config.Config
-	createOptions chartutil.CreateOptions
+	config   config.Config
+	resource *resource.Resource
+	options  chartutil.CreateOptions
 }
 
-var (
-	_ plugin.CreateAPISubcommand = &createAPISubcommand{}
-	_ cmdutil.RunOptions         = &createAPISubcommand{}
-)
-
-// UpdateContext define plugin context
-func (p createAPISubcommand) UpdateContext(ctx *plugin.Context) {
-	ctx.Description = `Scaffold a Kubernetes API that is backed by a Helm chart.
+func (p *createAPISubcommand) UpdateMetadata(cliMeta plugin.CLIMetadata, subcmdMeta *plugin.SubcommandMetadata) {
+	subcmdMeta.Description = `Scaffold a Kubernetes API that is backed by a Helm chart.
 `
-	ctx.Examples = fmt.Sprintf(`  $ %s create api \
+	subcmdMeta.Examples = fmt.Sprintf(`  $ %s create api \
       --group=apps --version=v1alpha1 \
       --kind=AppService
 
-  $ %s create api \
+  $ %[1]s create api \
       --group=apps --version=v1alpha1 \
       --kind=AppService \
       --helm-chart=myrepo/app
 
-  $ %s create api \
+  $ %[1]s create api \
       --helm-chart=myrepo/app
 
-  $ %s create api \
+  $ %[1]s create api \
       --helm-chart=myrepo/app \
       --helm-chart-version=1.2.3
 
-  $ %s create api \
+  $ %[1]s create api \
       --helm-chart=app \
       --helm-chart-repo=https://charts.mycompany.com/
 
-  $ %s create api \
+  $ %[1]s create api \
       --helm-chart=app \
       --helm-chart-repo=https://charts.mycompany.com/ \
       --helm-chart-version=1.2.3
 
-  $ %s create api \
+  $ %[1]s create api \
       --helm-chart=/path/to/local/chart-directories/app/
 
-  $ %s create api \
+  $ %[1]s create api \
       --helm-chart=/path/to/local/chart-archives/app-1.2.3.tgz
-`,
-		ctx.CommandName,
-		ctx.CommandName,
-		ctx.CommandName,
-		ctx.CommandName,
-		ctx.CommandName,
-		ctx.CommandName,
-		ctx.CommandName,
-		ctx.CommandName,
-	)
+`, cliMeta.CommandName)
 }
 
-const (
-	groupFlag            = "group"
-	versionFlag          = "version"
-	kindFlag             = "kind"
-	helmChartFlag        = "helm-chart"
-	helmChartRepoFlag    = "helm-chart-repo"
-	helmChartVersionFlag = "helm-chart-version"
-	crdVersionFlag       = "crd-version"
-
-	crdVersionV1      = "v1"
-	crdVersionV1beta1 = "v1beta1"
-)
-
 // BindFlags will set the flags for the plugin
 func (p *createAPISubcommand) BindFlags(fs *pflag.FlagSet) {
-	p.createOptions = chartutil.CreateOptions{}
 	fs.SortFlags = false
 
-	fs.StringVar(&p.createOptions.GVK.Group, groupFlag, "", "resource group")
```

Comment:
Are these flags still working on for the command?

Reply:
These flags (group, version and kind) are binded by the CLI and already configured in the injected resource.